kano/decorators.py

```diff
@@ -0,0 +1,51 @@
+# decorators.py
+#
+# Copyright (C) 2015 Kano Computing Ltd.
+# License: http://www.gnu.org/licenses/gpl-2.0.txt GNU GPL v2
+#
+# Decorators used to simplify control and ease modularity
+
+
+import os
+import sys
+from kano.logging import logger
+
+ERR_ROOT_PERMISSIONS_REQ = -1
+
+
+def require_root(exit_on_failure=False, verbose=False):
+    '''
+    Generates decorator to enforce root permissions
+
+    NB: must be called when used as decorator, i.e.
+        @require_root()
```

Comment:
For this one, yes - the `require_root` function is actually a decorator generator, not a decorator itself so needs to be called with parentheses. This is explained better in [this Stack Overflow thread](http://stackoverflow.com/questions/5929107/python-decorators-with-parameters#25827070).